diff --git a/src/main.rs b/src/main.rs
index 76cdfaa..1597e50 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -29,7 +29,7 @@ fn self_elevate() -> ! {
     use std::os::unix::process::CommandExt;
 
     let mut cmd = std::process::Command::new("sudo");
-    cmd.args(std::env::args());
+    cmd.arg("-A").args(std::env::args());
     debug!("{:?}", cmd);
     let err = cmd.exec();
     panic!("{}", err);
diff --git a/src/nixos.rs b/src/nixos.rs
index 32075d4..58c86db 100644
--- a/src/nixos.rs
+++ b/src/nixos.rs
@@ -40,7 +40,7 @@ impl OsRebuildArgs {
 
         // TODO: add a  .maybe_arg to CommandBuilder
         // so that I can do .maybe_arg( Option<T> )
-        let sudo_args: &[_] = if use_sudo { &["sudo"] } else { &[] };
+        let sudo_args: &[_] = if use_sudo { &["sudo", "-A"] } else { &[] };
 
         let hostname = match &self.hostname {
             Some(h) => h.to_owned(),